packages/node_modules/samples/browser-call-with-screenshare/app.js

```diff
@@ -10,67 +10,224 @@
-let activeMeeting, remoteShareStream, webex;
 
-// First, let's wire our form fields up to localStorage so we don't have to
-// retype things everytime we reload the page.
+let redirect_uri;
+let activeMeeting;
+let remoteShareStream;
+let isToken;
+let isOAuth;
+let webex;
+
+function getOAuthRedirectURI() {
+  // This is one way you might programattically determine your redirect_uri
+  // depending on where you've deployed your app, but you're probably better off
+  // having development/staging/production builds and injecting directly from the
+  // environment.
+  redirect_uri = `${window.location.protocol}//${window.location.host}`;
+
+  if (window.location.pathname) {
+    redirect_uri += window.location.pathname;
+  }
+}
 
-[
-  'access-token',
-  'invitee'
-].forEach((id) => {
-  const el = document.getElementById(id);
 
-  el.value = localStorage.getItem(id);
-  el.addEventListener('change', (event) => {
-    localStorage.setItem(id, event.target.value);
+// OAuth section
+if (window.location.hash) {
+  // Hash is created which gives us the access token,
+  // which will be used in the SDK initialization process
+  initializeWebexOAuthVersion();
+}
+
+// Initialize Webex to authorize through OAuth
+function initializeWebexOAuthVersion() {
+  // Get's redirect URI for OAuth option
+  getOAuthRedirectURI();
+
+  // eslint-disable-next-line no-multi-assign
+  webex = window.webex = Webex.init({
+    config: {
+      logger: {
+        level: 'debug'
+      },
+      meetings: {
+        reconnection: {
+          enabled: true
+        }
+      },
+      // Any other sdk config we need
+      credentials: {
+        client_id: 'C7c3f1143a552d88d40b2afff87600c366c830850231597fb6c1c1e28a5110a4f',
+        redirect_uri,
+        scope: 'spark:all spark:kms'
+      }
+    }
   });
-});
+}
 
-// There's a few different events that'll let us know we should initialize
-// Webex and start listening for incoming calls, so we'll wrap a few things
-// up in a function.
-function connect() {
-  return new Promise((resolve) => {
-    if (!webex) {
-      // eslint-disable-next-line no-multi-assign
-      webex = window.webex = Webex.init({
-        config: {
-          logger: {
-            level: 'debug'
-          },
-          meetings: {
-            reconnection: {
-              enabled: true
-            }
-          }
-          // Any other sdk config we need
-        },
+// Creates OAuth Form
+function createOAuthForm() {
+  // Present and hide correct form
+  document.getElementById('token-auth').style.display = 'none';
+  document.getElementById('oauth').style.display = 'block';
+
+  // Now, we adding event handler to login button to initiate login when clicked
+  document.getElementById('login').addEventListener('click', (event) => {
+    // Let's make sure we don't reload the page when we submit the form
+    event.preventDefault();
+
+    // Initialize the Webex
+    initializeWebexOAuthVersion();
```

Comment:
We don't need to wait until login is pressed to create the webex oauth instance. That is why the error on line 95 is throwing.